cardano-cli/test/cardano-cli-test-lib/Test/Cardano/CLI/Util.hs

```diff
@@ -196,6 +198,13 @@ createFiles = IO.unsafePerformIO $ do
   value <- IO.lookupEnv "CREATE_GOLDEN_FILES"
   return $ value == Just "1"
 
+-- | Asserts that the given directory is missing.
+assertDirectoryMissing :: (MonadTest m, MonadIO m, HasCallStack) => FilePath -> m ()
+assertDirectoryMissing dir = GHC.withFrozenCallStack $ do
+  exists <- H.evalIO $ IO.doesDirectoryExist dir
+  when exists $ H.failWithCustom GHC.callStack Nothing (dir <> " should not have been created.")
```

Comment:
Your solution is fine, but I think this may be more readable:
```suggestion
  when exists $ do
    H.note_ $ dir <> " should not have been created."
    H.failure
```

Reply:
I pretty much copied it from here: [https://hackage.haskell.org/package/hedgehog-extras-0.6.1.0/docs/src/Hedgehog.Extras.Test.File.html#assertFileMissing](https://hackage.haskell.org/package/hedgehog-extras-0.6.1.0/docs/src/Hedgehog.Extras.Test.File.html#assertFileMissing).

Maybe I should make a comment giving it credit, in retrospective, and/or make a PR to `hedgehog-extras`.
I suspect, even though your suggestion is indeed more readable, it may give worse error message when the test fails, and I don't know how it plays with `withFrozenCallStack`. I don't mind either way.

Reply:
Oh, wait, hedgehog-extras belongs to IOG?! I am definitely making a PR there.

Reply:
https://github.com/input-output-hk/hedgehog-extras/pull/64